src/main/java/taiga/api/APIWrapperBase.java

```diff
@@ -2,23 +2,30 @@
 
 import com.google.gson.Gson;
 import com.google.gson.JsonIOException;
+
 import java.net.URI;
 import java.net.URISyntaxException;
 import java.net.http.HttpRequest;
 import java.net.http.HttpResponse;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.Semaphore;
 import settings.Settings;
 import taiga.model.auth.RefreshResponse;
 import taiga.model.auth.Tokens;
 import taiga.util.AuthTokenSingleton;
+import java.util.concurrent.TimeUnit;
 import taiga.util.HTTPClientSingleton;
 import taiga.util.TokenStore;
+import java.util.Map;
 
 public abstract class APIWrapperBase {
 
     private String apiBaseURL;
     private final String apiEndpoint;
+    private Map<String, Semaphore> semaphores = new ConcurrentHashMap<>();
+    private static final int MAX_CONCURRENT_REQUESTS_NUMBER = 100;
```

Comment:
Seems like a magic number, any reasoning for choosing 100? If it works, it's fine, but it would be good to have additional reasoning on why this works.

Reply:
Here is an explanation for why I chose 100
https://stackoverflow.com/questions/68781943/appropriate-max-concurrent-streams-value-of-http2